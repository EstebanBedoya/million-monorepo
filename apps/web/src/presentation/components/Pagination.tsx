'use client';

interface PaginationProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
  hasNext: boolean;
  hasPrev: boolean;
}

export function Pagination({ 
  currentPage, 
  totalPages, 
  onPageChange, 
  hasNext, 
  hasPrev 
}: PaginationProps) {
  // Generate page numbers to display
  const getPageNumbers = () => {
    const pages: (number | string)[] = [];
    const maxVisible = 5; // Maximum number of page buttons to show

    if (totalPages <= maxVisible) {
      // Show all pages if total is less than max
      for (let i = 1; i <= totalPages; i++) {
        pages.push(i);
      }
    } else {
      // Always show first page
      pages.push(1);

      if (currentPage > 3) {
        pages.push('...');
      }

      // Show pages around current page
      const start = Math.max(2, currentPage - 1);
      const end = Math.min(totalPages - 1, currentPage + 1);

      for (let i = start; i <= end; i++) {
        pages.push(i);
      }

      if (currentPage < totalPages - 2) {
        pages.push('...');
      }

      // Always show last page
      pages.push(totalPages);
    }

    return pages;
  };

  if (totalPages <= 1) {
    return null;
  }

  return (
    <nav className="flex items-center justify-center gap-2 mt-8" role="navigation" aria-label="Pagination">
      {/* Previous button */}
      <button
        onClick={() => onPageChange(currentPage - 1)}
        disabled={!hasPrev}
        className={`
          px-4 py-2 rounded-md font-medium transition-all duration-200
          ${hasPrev 
            ? 'bg-card border border-border text-foreground hover:border-accent hover:text-accent' 
            : 'bg-secondary/10 text-secondary cursor-not-allowed'
          }
        `}
        aria-label="Previous page"
      >
        <svg 
          className="w-5 h-5" 
          fill="none" 
          stroke="currentColor" 
          viewBox="0 0 24 24"
          aria-hidden="true"
        >
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15 19l-7-7 7-7" />
        </svg>
      </button>

      {/* Page numbers */}
      <div className="flex items-center gap-1">
        {getPageNumbers().map((page, index) => {
          if (page === '...') {
            return (
              <span 
                key={`ellipsis-${index}`} 
                className="px-3 py-2 text-secondary"
                aria-hidden="true"
              >
                ...
              </span>
            );
          }

          const pageNumber = page as number;
          const isActive = pageNumber === currentPage;

          return (
            <button
              key={pageNumber}
              onClick={() => onPageChange(pageNumber)}
              className={`
                px-4 py-2 rounded-md font-medium transition-all duration-200
                ${isActive
                  ? 'bg-accent text-white shadow-sm'
                  : 'bg-card border border-border text-foreground hover:border-accent hover:text-accent'
                }
              `}
              aria-label={`Page ${pageNumber}`}
              aria-current={isActive ? 'page' : undefined}
            >
              {pageNumber}
            </button>
          );
        })}
      </div>

      {/* Next button */}
      <button
        onClick={() => onPageChange(currentPage + 1)}
        disabled={!hasNext}
        className={`
          px-4 py-2 rounded-md font-medium transition-all duration-200
          ${hasNext 
            ? 'bg-card border border-border text-foreground hover:border-accent hover:text-accent' 
            : 'bg-secondary/10 text-secondary cursor-not-allowed'
          }
        `}
        aria-label="Next page"
      >
        <svg 
          className="w-5 h-5" 
          fill="none" 
          stroke="currentColor" 
          viewBox="0 0 24 24"
          aria-hidden="true"
        >
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 5l7 7-7 7" />
        </svg>
      </button>
    </nav>
  );
}

