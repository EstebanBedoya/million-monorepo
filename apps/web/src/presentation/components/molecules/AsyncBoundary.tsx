'use client';

import { ReactNode } from 'react';
import { PropertyListSkeleton } from './PropertyCardSkeleton';
import { EmptyState } from './EmptyState';
import { ErrorState } from './ErrorState';

export interface AsyncBoundaryProps {
  children: ReactNode;
  loading?: boolean;
  error?: Error | string | null;
  isEmpty?: boolean;
  emptyTitle?: string;
  emptyMessage?: string;
  onRetry?: () => void;
  onClearFilters?: () => void;
  skeletonCount?: number;
  className?: string;
}

export function AsyncBoundary({
  children,
  loading = false,
  error = null,
  isEmpty = false,
  emptyTitle,
  emptyMessage,
  onRetry,
  onClearFilters,
  skeletonCount = 12,
  className
}: AsyncBoundaryProps) {
  // Show loading state
  if (loading) {
    return <PropertyListSkeleton count={skeletonCount} className={className} />;
  }

  // Show error state
  if (error) {
    return (
      <ErrorState 
        error={error}
        onRetry={onRetry}
        className={className}
      />
    );
  }

  // Show empty state
  if (isEmpty) {
    return (
      <EmptyState 
        title={emptyTitle}
        message={emptyMessage}
        onReset={onClearFilters}
        className={className}
      />
    );
  }

  // Show content
  return <>{children}</>;
}
