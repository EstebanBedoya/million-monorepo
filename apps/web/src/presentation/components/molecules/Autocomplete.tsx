'use client';

import { useState, useRef, useEffect } from 'react';
import { Search, ChevronDown, Loader2 } from 'lucide-react';
import { Input } from '@/presentation/components/atoms/Input';
import { cn } from '@/utils/cn';

export interface AutocompleteOption {
  value: string;
  label: string;
  description?: string;
}

interface AutocompleteProps {
  options: AutocompleteOption[];
  value: string;
  onChange: (value: string) => void;
  onSearch?: (query: string) => void;
  placeholder?: string;
  emptyMessage?: string;
  loading?: boolean;
  error?: boolean;
  disabled?: boolean;
  className?: string;
}

export function Autocomplete({
  options,
  value,
  onChange,
  onSearch,
  placeholder = 'Search...',
  emptyMessage = 'No results found',
  loading = false,
  error = false,
  disabled = false,
  className,
}: AutocompleteProps) {
  const [isOpen, setIsOpen] = useState(false);
  const [searchQuery, setSearchQuery] = useState('');
  const [highlightedIndex, setHighlightedIndex] = useState(-1);
  const containerRef = useRef<HTMLDivElement>(null);
  const inputRef = useRef<HTMLInputElement>(null);

  const selectedOption = options.find((opt) => opt.value === value);
  const displayValue = selectedOption?.label || '';

  const filteredOptions = options.filter((option) =>
    option.label.toLowerCase().includes(searchQuery.toLowerCase())
  );

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (containerRef.current && !containerRef.current.contains(event.target as Node)) {
        setIsOpen(false);
        setSearchQuery('');
      }
    };

    document.addEventListener('mousedown', handleClickOutside);
    return () => document.removeEventListener('mousedown', handleClickOutside);
  }, []);

  useEffect(() => {
    if (onSearch && searchQuery) {
      const timeoutId = setTimeout(() => {
        onSearch(searchQuery);
      }, 300);
      return () => clearTimeout(timeoutId);
    }
  }, [searchQuery, onSearch]);

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const query = e.target.value;
    setSearchQuery(query);
    setIsOpen(true);
    setHighlightedIndex(-1);
  };

  const handleOptionClick = (option: AutocompleteOption) => {
    onChange(option.value);
    setIsOpen(false);
    setSearchQuery('');
    setHighlightedIndex(-1);
  };

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (!isOpen && (e.key === 'ArrowDown' || e.key === 'Enter')) {
      setIsOpen(true);
      return;
    }

    if (!isOpen) return;

    switch (e.key) {
      case 'ArrowDown':
        e.preventDefault();
        setHighlightedIndex((prev) =>
          prev < filteredOptions.length - 1 ? prev + 1 : prev
        );
        break;
      case 'ArrowUp':
        e.preventDefault();
        setHighlightedIndex((prev) => (prev > 0 ? prev - 1 : -1));
        break;
      case 'Enter':
        e.preventDefault();
        if (highlightedIndex >= 0 && filteredOptions[highlightedIndex]) {
          handleOptionClick(filteredOptions[highlightedIndex]);
        }
        break;
      case 'Escape':
        setIsOpen(false);
        setSearchQuery('');
        setHighlightedIndex(-1);
        break;
    }
  };

  const handleFocus = () => {
    setIsOpen(true);
  };

  return (
    <div ref={containerRef} className={cn('relative', className)}>
      <div className="relative">
        <Input
          ref={inputRef}
          type="text"
          value={isOpen ? searchQuery : displayValue}
          onChange={handleInputChange}
          onKeyDown={handleKeyDown}
          onFocus={handleFocus}
          placeholder={placeholder}
          disabled={disabled}
          error={error}
          icon={loading ? <Loader2 className="w-4 h-4 animate-spin" /> : <Search className="w-4 h-4" />}
          className="pr-10"
        />
        <button
          type="button"
          onClick={() => setIsOpen(!isOpen)}
          disabled={disabled}
          className="absolute right-3 top-1/2 -translate-y-1/2 text-secondary hover:text-foreground transition-colors disabled:opacity-50"
        >
          <ChevronDown
            className={cn(
              'w-4 h-4 transition-transform',
              isOpen && 'transform rotate-180'
            )}
          />
        </button>
      </div>

      {isOpen && (
        <div className="absolute z-50 w-full mt-1 bg-card border border-border rounded-md shadow-lg max-h-60 overflow-auto">
          {loading ? (
            <div className="p-4 text-center text-secondary">
              <Loader2 className="w-5 h-5 animate-spin mx-auto mb-2" />
              <p className="text-sm">Loading...</p>
            </div>
          ) : filteredOptions.length === 0 ? (
            <div className="p-4 text-center text-secondary text-sm">
              {emptyMessage}
            </div>
          ) : (
            <ul className="py-1">
              {filteredOptions.map((option, index) => (
                <li key={option.value}>
                  <button
                    type="button"
                    onClick={() => handleOptionClick(option)}
                    className={cn(
                      'w-full px-4 py-2 text-left hover:bg-accent/10 transition-colors',
                      highlightedIndex === index && 'bg-accent/10',
                      value === option.value && 'bg-accent/20 font-medium'
                    )}
                  >
                    <div className="text-sm text-foreground">{option.label}</div>
                    {option.description && (
                      <div className="text-xs text-secondary mt-0.5">
                        {option.description}
                      </div>
                    )}
                  </button>
                </li>
              ))}
            </ul>
          )}
        </div>
      )}
    </div>
  );
}

