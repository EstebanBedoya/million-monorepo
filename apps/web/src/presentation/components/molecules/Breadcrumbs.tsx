'use client';

import Link from 'next/link';
import { ChevronRight } from 'lucide-react';
import { Icon } from '../atoms/Icon';

export interface BreadcrumbItem {
  label: string;
  href?: string;
}

export interface BreadcrumbsProps {
  items: BreadcrumbItem[];
  className?: string;
}

export const Breadcrumbs = ({ items, className }: BreadcrumbsProps) => {
  return (
    <nav 
      aria-label="Breadcrumb" 
      className={className}
    >
      <ol className="flex items-center space-x-2 flex-wrap">
        {items.map((item, index) => {
          const isLast = index === items.length - 1;
          
          return (
            <li key={index} className="flex items-center">
              {index > 0 && (
                <Icon 
                  icon={ChevronRight} 
                  size="sm" 
                  className="text-secondary mx-2" 
                  aria-hidden={true}
                />
              )}
              
              {isLast || !item.href ? (
                <span 
                  className="text-sm font-medium text-foreground"
                  aria-current={isLast ? 'page' : undefined}
                >
                  {item.label}
                </span>
              ) : (
                <Link 
                  href={item.href}
                  className="text-sm text-secondary hover:text-accent transition-colors"
                >
                  {item.label}
                </Link>
              )}
            </li>
          );
        })}
      </ol>
    </nav>
  );
};

