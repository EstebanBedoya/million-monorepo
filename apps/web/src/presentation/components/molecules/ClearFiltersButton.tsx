'use client';

import { X } from 'lucide-react';
import { Button } from '../atoms/Button';
import { Icon } from '../atoms/Icon';

export interface ClearFiltersButtonProps {
  onClick: () => void;
  className?: string;
}

export const ClearFiltersButton = ({ onClick, className }: ClearFiltersButtonProps) => {
  return (
    <Button
      variant="secondary"
      onClick={onClick}
      className={`w-full lg:w-auto ${className}`}
      aria-label="Clear all filters"
    >
      <Icon icon={X} size="sm" />
      Clear All Filters
    </Button>
  );
};
