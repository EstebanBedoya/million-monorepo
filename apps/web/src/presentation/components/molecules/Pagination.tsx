'use client';

import { ChevronLeft, ChevronRight } from 'lucide-react';
import { Button } from '../atoms/Button';
import { Icon } from '../atoms/Icon';

export interface PaginationProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
  hasNext: boolean;
  hasPrev: boolean;
  className?: string;
}

export const Pagination = ({
  currentPage,
  totalPages,
  onPageChange,
  hasNext,
  hasPrev,
  className
}: PaginationProps) => {
  // Don't render if there's only one page or less
  if (totalPages <= 1) {
    return null;
  }

  const getPageNumbers = () => {
    const pages: (number | string)[] = [];
    const maxVisiblePages = 5;

    if (totalPages <= maxVisiblePages) {
      // Show all pages if total is small
      for (let i = 1; i <= totalPages; i++) {
        pages.push(i);
      }
    } else {
      // Always show first page
      pages.push(1);

      if (currentPage > 3) {
        pages.push('...');
      }

      // Show pages around current page
      const start = Math.max(2, currentPage - 1);
      const end = Math.min(totalPages - 1, currentPage + 1);

      for (let i = start; i <= end; i++) {
        pages.push(i);
      }

      if (currentPage < totalPages - 2) {
        pages.push('...');
      }

      // Always show last page
      if (totalPages > 1) {
        pages.push(totalPages);
      }
    }

    return pages;
  };

  const pageNumbers = getPageNumbers();

  return (
    <nav
      className={className}
      role="navigation"
      aria-label="Pagination"
    >
      <div className="flex items-center justify-center space-x-2">
        <Button
          variant="outline"
          size="sm"
          onClick={() => onPageChange(currentPage - 1)}
          disabled={!hasPrev}
          aria-label="Previous page"
        >
          <Icon icon={ChevronLeft} size="sm" />
          Previous
        </Button>

        <div className="flex items-center space-x-1">
          {pageNumbers.map((page, index) => {
            if (page === '...') {
              return (
                <span
                  key={`ellipsis-${index}`}
                  className="px-3 py-2 text-sm text-secondary"
                  aria-hidden="true"
                >
                  ...
                </span>
              );
            }

            const pageNum = page as number;
            const isCurrentPage = pageNum === currentPage;

            return (
              <Button
                key={pageNum}
                variant={isCurrentPage ? 'primary' : 'outline'}
                size="sm"
                onClick={() => onPageChange(pageNum)}
                aria-label={`Page ${pageNum}`}
                aria-current={isCurrentPage ? 'page' : undefined}
                className={isCurrentPage ? 'pointer-events-none' : ''}
              >
                {pageNum}
              </Button>
            );
          })}
        </div>

        <Button
          variant="outline"
          size="sm"
          onClick={() => onPageChange(currentPage + 1)}
          disabled={!hasNext}
          aria-label="Next page"
        >
          Next
          <Icon icon={ChevronRight} size="sm" />
        </Button>
      </div>
    </nav>
  );
};
