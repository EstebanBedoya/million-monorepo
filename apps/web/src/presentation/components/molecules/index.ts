export { SearchInput } from './SearchInput';
export { PriceRange } from './PriceRange';
export { PriceSuggestions } from './PriceSuggestions';
export { PropertyTypeSelect } from './PropertyTypeSelect';
export { PropertyDetails } from './PropertyDetails';
export { LocationInfo } from './LocationInfo';
export { ClearFiltersButton } from './ClearFiltersButton';
export { PageButton } from './PageButton';
export { PageNumberButton } from './PageNumberButton';
export { EmptyState } from './EmptyState';
export { PropertyCardSkeleton, PropertyListSkeleton } from './PropertyCardSkeleton';
export { PaginationSkeleton } from './PaginationSkeleton';
export { ThemeToggle } from './ThemeToggle';
export { AsyncBoundary } from './AsyncBoundary';
export { ErrorState } from './ErrorState';
export { LoadingState } from './LoadingState';
export { Breadcrumbs } from './Breadcrumbs';
