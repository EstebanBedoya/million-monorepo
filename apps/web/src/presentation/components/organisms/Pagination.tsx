'use client';

import { PageButton } from '../molecules/PageButton';
import { PageNumberButton } from '../molecules/PageNumberButton';

interface PaginationProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
  hasNext: boolean;
  hasPrev: boolean;
}

export function Pagination({ 
  currentPage, 
  totalPages, 
  onPageChange, 
  hasNext, 
  hasPrev 
}: PaginationProps) {
  // Generate page numbers to display
  const getPageNumbers = () => {
    const pages: (number | string)[] = [];
    const maxVisible = 5; // Maximum number of page buttons to show

    if (totalPages <= maxVisible) {
      // Show all pages if total is less than max
      for (let i = 1; i <= totalPages; i++) {
        pages.push(i);
      }
    } else {
      // Always show first page
      pages.push(1);

      if (currentPage > 3) {
        pages.push('...');
      }

      // Show pages around current page
      const start = Math.max(2, currentPage - 1);
      const end = Math.min(totalPages - 1, currentPage + 1);

      for (let i = start; i <= end; i++) {
        pages.push(i);
      }

      if (currentPage < totalPages - 2) {
        pages.push('...');
      }

      // Always show last page
      pages.push(totalPages);
    }

    return pages;
  };

  if (totalPages <= 1) {
    return null;
  }

  return (
    <nav className="flex items-center justify-center gap-2 mt-8" role="navigation" aria-label="Pagination">
      {/* Previous button */}
      <PageButton
        direction="prev"
        disabled={!hasPrev}
        onClick={() => onPageChange(currentPage - 1)}
      />

      {/* Page numbers */}
      <div className="flex items-center gap-1">
        {getPageNumbers().map((page, index) => {
          if (page === '...') {
            return (
              <span 
                key={`ellipsis-${index}`} 
                className="px-3 py-2 text-secondary"
                aria-hidden="true"
              >
                ...
              </span>
            );
          }

          const pageNumber = page as number;
          const isActive = pageNumber === currentPage;

          return (
            <PageNumberButton
              key={pageNumber}
              page={pageNumber}
              isActive={isActive}
              onClick={() => onPageChange(pageNumber)}
            />
          );
        })}
      </div>

      {/* Next button */}
      <PageButton
        direction="next"
        disabled={!hasNext}
        onClick={() => onPageChange(currentPage + 1)}
      />
    </nav>
  );
}
