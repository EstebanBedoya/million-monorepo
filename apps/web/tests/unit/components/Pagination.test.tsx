import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Pagination } from '../../../src/presentation/components/molecules/Pagination';

describe('Pagination', () => {
  const mockOnPageChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should not render when totalPages is 1 or less', () => {
    const { container } = render(
      <Pagination 
        currentPage={1}
        totalPages={1}
        onPageChange={mockOnPageChange}
        hasNext={false}
        hasPrev={false}
      />
    );

    expect(container).toBeEmptyDOMElement();
  });

  it('should render page numbers correctly', () => {
    render(
      <Pagination 
        currentPage={1}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={false}
      />
    );

    expect(screen.getByRole('button', { name: 'Page 1' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Page 2' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Page 5' })).toBeInTheDocument();
  });

  it('should highlight current page', () => {
    render(
      <Pagination 
        currentPage={3}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={true}
      />
    );

    const currentPageButton = screen.getByRole('button', { name: 'Page 3', current: 'page' });
    expect(currentPageButton).toBeInTheDocument();
  });

  it('should disable previous button when hasPrev is false', () => {
    render(
      <Pagination 
        currentPage={1}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={false}
      />
    );

    const prevButton = screen.getByRole('button', { name: 'Previous page' });
    expect(prevButton).toBeDisabled();
  });

  it('should disable next button when hasNext is false', () => {
    render(
      <Pagination 
        currentPage={5}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={false}
        hasPrev={true}
      />
    );

    const nextButton = screen.getByRole('button', { name: 'Next page' });
    expect(nextButton).toBeDisabled();
  });

  it('should call onPageChange when page number is clicked', async () => {
    render(
      <Pagination 
        currentPage={1}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={false}
      />
    );

    const page3Button = screen.getByRole('button', { name: 'Page 3' });
    await userEvent.click(page3Button);

    expect(mockOnPageChange).toHaveBeenCalledWith(3);
  });

  it('should call onPageChange when previous button is clicked', async () => {
    render(
      <Pagination 
        currentPage={3}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={true}
      />
    );

    const prevButton = screen.getByRole('button', { name: 'Previous page' });
    await userEvent.click(prevButton);

    expect(mockOnPageChange).toHaveBeenCalledWith(2);
  });

  it('should call onPageChange when next button is clicked', async () => {
    render(
      <Pagination 
        currentPage={3}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={true}
      />
    );

    const nextButton = screen.getByRole('button', { name: 'Next page' });
    await userEvent.click(nextButton);

    expect(mockOnPageChange).toHaveBeenCalledWith(4);
  });

  it('should show ellipsis for large page counts', () => {
    render(
      <Pagination 
        currentPage={10}
        totalPages={20}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={true}
      />
    );

    const ellipsis = screen.getAllByText('...');
    expect(ellipsis.length).toBeGreaterThan(0);
  });

  it('should have proper navigation attributes', () => {
    render(
      <Pagination 
        currentPage={2}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={true}
      />
    );

    const nav = screen.getByRole('navigation', { name: 'Pagination' });
    expect(nav).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <Pagination 
        currentPage={2}
        totalPages={5}
        onPageChange={mockOnPageChange}
        hasNext={true}
        hasPrev={true}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

