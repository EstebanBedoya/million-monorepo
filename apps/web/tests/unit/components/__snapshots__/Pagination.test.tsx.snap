// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match snapshot 1`] = `
<nav
  aria-label="Pagination"
  role="navigation"
>
  <div
    class="flex items-center justify-center space-x-2"
  >
    <button
      aria-label="Previous page"
      class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-8 px-3 text-sm"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-left w-4 h-4 text-current"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m15 18-6-6 6-6"
        />
      </svg>
      Previous
    </button>
    <div
      class="flex items-center space-x-1"
    >
      <button
        aria-label="Page 1"
        class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-8 px-3 text-sm"
      >
        1
      </button>
      <button
        aria-current="page"
        aria-label="Page 2"
        class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed bg-accent text-white hover:bg-accent/90 focus:ring-accent h-8 px-3 text-sm pointer-events-none"
      >
        2
      </button>
      <button
        aria-label="Page 3"
        class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-8 px-3 text-sm"
      >
        3
      </button>
      <button
        aria-label="Page 4"
        class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-8 px-3 text-sm"
      >
        4
      </button>
      <button
        aria-label="Page 5"
        class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-8 px-3 text-sm"
      >
        5
      </button>
    </div>
    <button
      aria-label="Next page"
      class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-8 px-3 text-sm"
    >
      Next
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-right w-4 h-4 text-current"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m9 18 6-6-6-6"
        />
      </svg>
    </button>
  </div>
</nav>
`;
